### ISSUE TITLE
Duplicated Resnet50 code and weight files

### ISSUE THREAD

**Issue by NivekNey** [keras-team/keras-applications#74]
These are for `include_top=False`, the weight files are not pointing to the same weight file. They also differ in terms of layer naming. Is there a reason to keep them both? Also noticed `tf.keras.applications` is pointing to `resnet50.ResNet50`.

* For `resnet50.ResNet50`:
https://github.com/keras-team/keras-applications/blob/d506dc82d0/keras_applications/resnet50.py#L27-L29
```
md5sum resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5
a268eb855778b3df3c7506639542a6af  resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5
```

* For `resnet.ResNet50`/`resnet_common.ResNet50`:
https://github.com/keras-team/keras-applications/blob/d506dc82d0/keras_applications/resnet_common.py#L40-L42
```
md5sum resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5
4d473c1dd8becc155b73f8504c6f6626  resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5
```

**Comment by taehoonlee** [comment]
@NivekNey, the `resnet50.ResNet50` is a legacy and `resnet.ResNet50`(=`resnet_common.ResNet50`) is factorized over the ResNet variations and new one. But the reason why we are keeping the weight file is not to break the current usages. For example, someone, who are thresholding of softmax values of `ResNet50`, may get the different results if we are replacing the `resnet50.ResNet50` with `resnet.ResNet50`.

**Comment by NivekNey** [comment]
Ok got it, I guess this is more a model zoo than, so it makes sense to keep both for people to choose/use. Thanks for the official reply!
